docs/content/tutorials/cluster.md

```diff
@@ -0,0 +1,305 @@
+```
+
+Also,
+
+- Place your Hadoop configuration XMLs (core-site.xml, hdfs-site.xml, yarn-site.xml,
+mapred-site.xml) on the classpath of your Druid nodes. You can do this by copying them into
+`conf/_common/`.
+
+## Configure cluster for Hadoop data loads
+
+Druid supports a variety of [realtime and batch data ingestion methods](../ingestion/overview.md). If you will
+be loading data from a Hadoop cluster, then at this point you should configure Druid to be aware
+of your cluster:
+
+- Update `druid.indexer.task.hadoopWorkingPath` in `conf/middleManager/runtime.properties` to
+a path on HDFS that you'd like to use for temporary files required during the indexing process.
+`druid.indexer.task.hadoopWorkingPath=/tmp/druid-indexing` is a common choice.
+
+- Place your Hadoop configuration XMLs (core-site.xml, hdfs-site.xml, yarn-site.xml,
+mapred-site.xml) on the classpath of your Druid nodes. You can do this by copying them into
+`conf/_common/core-site.xml`, `conf/_common/hdfs-site.xml`, and so on.
+
+Note that you don't need to use HDFS deep storage in order to load data from Hadoop. For example, if
+your cluster is running on Amazon Web Services, we recommend using S3 for deep storage even if you
+are loading data using Hadoop or Elastic MapReduce.
+
+## Configure addresses for Druid coordination
+
+In this simple cluster, you will deploy a single Druid Coordinator, a
+single Druid Overlord, a single ZooKeeper instance, and an embedded Derby metadata store on the same server.
+
+In `conf/_common/common.runtime.properties`, replace
+"zk.host.ip" with the IP address of the machine that runs your ZK instance:
+
+- `druid.zk.service.host`
+
+In `conf/_common/common.runtime.properties`, replace
+"metadata.store.ip" with the IP address of the machine that you will use as your metadata store:
+
+- `druid.metadata.storage.connector.connectURI`
+- `druid.metadata.storage.connector.host`
+
+```note-caution
+In production, we recommend running 2 servers, each running a Druid Coordinator
+and a Druid Overlord. We also recommend running a ZooKeeper cluster on its own dedicated hardware,
+as well as  replicated [metadata
+storage](http://druid.io/docs/latest/dependencies/metadata-storage.html) such as MySQL or
+PostgreSQL, on its own dedicated hardware.
+```
+
+## Tune Druid processes that serve queries
+
+Druid Historicals and MiddleManagers can be co-located on the same hardware. Both Druid processes benefit greatly from 
+being tuned to the hardware they run on. If you are using [r3.2xlarge](https://aws.amazon.com/ec2
+/instance-types/#r3) EC2 instances, or similar hardware, the configuration in the distribution is a
+reasonable starting point.
+
+If you are using different hardware, we recommend adjusting configurations for your specific
+hardware. The most commonly adjusted configurations are:
+
+- `-Xmx` and `-Xms`
+- `druid.server.http.numThreads`
+- `druid.processing.buffer.sizeBytes`
+- `druid.processing.numThreads`
```

Comment:
mention the catch about keeping -XX:MaxDirectMemory >= numThreads*sizeBytes or else druid process will fail to start.